textattack/search_methods/genetic_algorithm.py

```diff
@@ -104,23 +103,52 @@ def _perturb(self, pop_member, original_result, index=None):
                 )
                 return pop_member
 
-            num_replacements_per_word[idx] = 0
+            word_attr_list[idx] = 0
             iterations += 1
         return pop_member
 
     @abstractmethod
     def _crossover_operation(self, pop_member1, pop_member2):
-        """Actual operation for generating crossover between pop_member1 and
-        pop_member2.
+        """Actual operation that takes `pop_member1` text and `pop_member2`
+        text and mixes the two to generate crossover between `pop_member1` and
+        `pop_member2`.
 
         Args:
             pop_member1 (PopulationMember): The first population member.
             pop_member2 (PopulationMember): The second population member.
         Returns:
-            Tuple of `AttackedText` and `np.array` for new text and its corresponding `num_replacements_per_word`.
+            Tuple of `AttackedText` and `np.array` for new text and its corresponding `word_attr_list`.
         """
         raise NotImplementedError()
 
+    def _post_crossover_check(
+        self, new_text, parent_text1, parent_text2, original_text
+    ):
+        """Check if `new_text` that has been produced by performing crossover
+        between `parent_text1` and `parent_text2` aligns with the constraints.
+
+        Args:
+            new_text (AttackedText): Text produced by crossover operation
+            parent_text1 (AttackedText): Parent text of `new_text`
+            parent_text2 (AttackedText): Second parent text of `new_text`
+            original_text (AttackedText): Original text
+        Returns:
+            `True` if `new_text` meets the constraints. If otherwise, return `False`.
+        """
+        if "last_transformation" in new_text.attack_attrs:
```

Comment:
This constraint checking logic still seems confusing to me. Why do we not check constraints if the 2 parent texts haven't been previously transformed?

Reply:
If both parents texts haven't been transformed, then both are equivalent to the original text. Therefore, the child must also be equivalent to the original text, meaning the constraint check is not necessary. We could do the check, but checking constraint requires `last_transformation` to exist in the `attack_attrs` of the transformed text.